src/app/CommandResponderInterface.h

```diff
@@ -0,0 +1,64 @@
+/*
+ *    Copyright (c) 2024 Project CHIP Authors
+ *    All rights reserved.
+ *
+ *    Licensed under the Apache License, Version 2.0 (the "License");
+ *    you may not use this file except in compliance with the License.
+ *    You may obtain a copy of the License at
+ *
+ *        http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *    Unless required by applicable law or agreed to in writing, software
+ *    distributed under the License is distributed on an "AS IS" BASIS,
+ *    WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *    See the License for the specific language governing permissions and
+ *    limitations under the License.
+ */
+
+#pragma once
+
+#include <lib/core/Optional.h>
+#include <messaging/ExchangeHolder.h>
+#include <protocols/interaction_model/StatusCode.h>
+
+namespace chip {
+namespace app {
+
+/**
+ * Interface for sending InvokeResponseMessage(s).
+ *
+ * Provides information about the associated exchange context.
+ *
+ * Design Rationale: This interface enhances unit testability and allows applications to
+ * customize CommandResponder behavior. For example, it can stub out sending a response.
+ */
+class CommandResponderInterface
+{
+public:
+    virtual ~CommandResponderInterface() = default;
+
+    /* ExchangeContext related getter methods */
```

Comment:
This needs much better documentation.  The exchange involved is the one the InvokeRequestMessage was received on, presumably.  The subject descriptor is the one for the exchange presumably, right?  But do we in fact want to allow that to be configured separately from the exchange context (as opposed to having a method on this class that uses the overridable GetExchangeContext to implement GetSubjectDescriptor)?  Similar for GetAccessingFabricIndex().

GetGroupId() needs to document what its return value means.

Reply:
Right now, to give unit test some flexibility it is helpful if we provide this instead of having CommandHandler get it from exchange context.

I will add todo comment here to address this separately.